src/PrestaShopBundle/Entity/Repository/TabRepository.php

```diff
@@ -121,24 +121,27 @@ public function changeEnabledByModuleName($moduleName, $enabled)
     /**
      * @param string $className
      *
-     * @return int
+     * @return ?int
      *
-     * @throws \Doctrine\ORM\NoResultException
      * @throws \Doctrine\ORM\NonUniqueResultException
      */
-    public function getIdByClassName(string $className): int
+    public function getIdByClassName(string $className): ?int
     {
         if (!isset($this->cachedTabIds[$className])) {
-            $result = $this->createQueryBuilder('t')
-                ->select('t.id, t.className')
-                ->where('t.className = :className')
-                ->andWhere('t.id != 0')
-                ->setParameter('className', $className)
-                ->getQuery()
-                ->getSingleResult(Query::HYDRATE_ARRAY)
-            ;
-
-            $this->cachedTabIds[$result['className']] = (int) $result['id'];
+            try {
+                $result = $this->createQueryBuilder('t')
+                    ->select('t.id, t.className')
+                    ->where('t.className = :className')
+                    ->andWhere('t.id != 0')
+                    ->setParameter('className', $className)
+                    ->getQuery()
+                    ->getSingleResult(Query::HYDRATE_ARRAY)
+                ;
+
+                $this->cachedTabIds[$result['className']] = (int) $result['id'];
+            } catch (\Doctrine\ORM\NoResultException) {
```

Comment:
I think the use case you met is quite an exception to the usual behaviour So maybe e should catch the exception in the Twig component instead which implies:
- we don't expect this method not to return a value
- we prefer it to throw an exception when it doesn't find a value as it will force us to look into the problem and understand it instead of silently failing
- in our case we identified that when we generate the legacy token it's possible that the tab ID used for generation may not be found, but then we handle the fallback for this specific use case

Reply:
Actually in you error case I believe you can't find a mapping tab ID mostly because you don't even have a controller name to search for So maybe you should instead check if controller name is not empty before you search for the tab And you can use 0 or empty string as the fallback for your token generation